src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,152 @@
+# 
+# ## Arguments 
+# * `sim::Simulation{T}`: [`Simulation`](@ref) for which the [`WeightingPotential`] should be adapted.
+# * `contact_id::Int`: The `id` of the [`Contact`](@ref) at which the [`WeightingPotential`] should be adapted.
+# """
+function _adapt_weighting_potential_to_electric_potential_grid!(sim::Simulation, contact_id::Int)
+    @assert !ismissing(sim.electric_potential) "Electric potential missing"
+    if ismissing(sim.weighting_potentials[contact_id]) calculate_weighting_potential!(sim, contact_id, verbose = false) end
+    if sim.weighting_potentials[contact_id].grid != sim.electric_potential.grid
+        sim.weighting_potentials[contact_id] = sim.weighting_potentials[contact_id][sim.electric_potential.grid]
+        #apply_initial_state!(sim, WeightingPotential, contact_id, sim.electric_potential.grid)
+        update_till_convergence!(sim, WeightingPotential, contact_id)
+    end
+end
+
+"""
+    get_depletion_voltage( sim::Simulation{T}, contact_id::Int, potential_range::AbstractRange; kwargs... )::T
+
+Returns the potential (in V) needed at the [`Contact`](@ref) with id `contact_id`
+to fully deplete the detector in a given [`Simulation`](@ref). For this, all other
+contact potentials are set to `0` and the potential at the specified contact is
+increased or decreased according to the `potential_range`. The depletion voltage
+is set to the potential for which a previously undepleted detector becomes depleted,
+or for which a previously depleted detector becomes undepleted.
+
+## Arguments 
+* `sim::Simulation{T}`: [`Simulation`](@ref) for which the depletion voltage should be determined.
+* `contact_id::Int`: The `id` of the [`Contact`](@ref) at which the potential is applied.
+* `potential_range::AbstractRange`: Range of potentials to be tested.
+    
+## Keywords
+* `verbose::Bool = true`: Activate or deactivate additional info output. Default is `true`.
+
+## Example 
+```julia
+using SolidStateDetectors
+sim = Simulation(SSD_examples[:InvertedCoax])
+calculate_electric_potential!(sim)
+get_depletion_voltage(sim, 2, 1600:1:2500)
+```
+
+!!! warn
+    This method only works if the initial `sim` was calculated for a case in which the detector is fully depleted.
+
+!!! note
+    The accuracy of the result depends on the precision of the initial simulation.
+    
+See also [`is_depleted`](@ref).
+"""
+function get_depletion_voltage(sim::Simulation{T}, contact_id::Int,
+            potential_range::AbstractRange = range(extrema(broadcast(c -> abs(c.potential), sim.detector.contacts))..., length = 1001);
+            verbose = true)::T where {T <: AbstractFloat}
+    
+    @assert is_depleted(sim.point_types) "This method only works for fully depleted simulations. Please increase the potentials in the configuration file to a greater value."
+    @assert sum(potential_range.>=0)==length(potential_range) "Please use a strictly positive potential range"
```

Comment:
I screwed up and did not put this into a seperate commit. When allowing negative ranges, the calls to range parameters would need to be flipped in that case (going form first entry to last is wrong if range goes from -150:0. We would need to start at 0 and go down). On the other side it should not really make a difference, when using a purely positive range. Therefor I decided to restrict the range to positive potentials.

Reply:
But what if the potential needed to deplete the detector (e.g. at the p++ contact of a p-type detector) is negative? 

Can you maybe check if the values are negative and then modify the range with a step of -1 if the range should reach into the negative?

Reply:
Sure, I forgot about doping